presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -924,6 +945,29 @@ public void dropTable(HiveIdentity identity, String databaseName, String tableNa
         }
     }
 
+    private static void deleteTableFilesIfNeeded(HdfsContext context, HdfsEnvironment hdfsEnvironment, String databaseName, String tableName, Path tablePath)
+    {
+        try {
+            if (isS3FileSystem(context, hdfsEnvironment, tablePath)) {
+                FileSystem fs = hdfsEnvironment.getFileSystem(context, tablePath);
+                if (fs.listFiles(tablePath, true).hasNext()) {
+                    log.warn("Table \"%s\".\"%s\" dropped, but files exist in '%s'", databaseName, tableName, tablePath.toUri());
+                    if (fs.delete(tablePath, true)) {
+                        log.warn("Deleted files for table \"%s\".\"%s\" in '%s' (Workaround for issue #1053)", databaseName, tableName, tablePath.toUri());
+                    }
+                }
+            }
+        }
+        catch (IOException | RuntimeException e) {
+            log.error(e, "Failed to delete files for table %s.%s: %s", databaseName, tableName, tablePath.toUri());
```

Comment:
Let's `warn` as in `GlueHiveMetastore`. We should use the same code (but copy it over for since it's short, rather than figuring out how to share it)